svm/src/account_loader.rs

```diff
@@ -213,29 +213,30 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
             let account = if solana_sdk::sysvar::instructions::check_id(key) {
                 construct_instructions_account(message)
             } else {
-                let instruction_account = u8::try_from(i)
-                    .map(|i| instruction_accounts.contains(&&i))
-                    .unwrap_or(false);
                 let (account_size, mut account, rent) = if let Some(account_override) =
                     account_overrides.and_then(|overrides| overrides.get(key))
                 {
                     (account_override.data().len(), account_override.clone(), 0)
-                } else if let Some(program) = (!instruction_account && !message.is_writable(i))
-                    .then_some(())
-                    .and_then(|_| loaded_programs.find(key))
-                {
-                    callbacks
-                        .get_account_shared_data(key)
-                        .ok_or(TransactionError::AccountNotFound)?;
-                    // Optimization to skip loading of accounts which are only used as
-                    // programs in top-level instructions and not passed as instruction accounts.
-                    let program_account = account_shared_data_from_program(&program);
-                    (program.account_size, program_account, 0)
                 } else {
```

Comment:
This optimization, which set program account data to empty, is not applicable
any more because we no longer prime program cache before loading the accounts. 

There is also a change of the reported execution errors when a transaction is
called for uninitialized or buffer program accounts. The original code will
throw "InvalidAccountData" error, while this PR will throw
"InvalidProgramForExecution" error. The reason is that original code will mark
the program account executable and failed on a later check when execute the
program. This PR will fail on `executable` check in this function early on.

Fortunately, this change won't affect consensus because program error are not
part of the consensus.